src/mmrelay/matrix_utils.py

```
@@ -889,8 +994,26 @@ async def connect_matrix(passed_config=None):
+                logger.error(
+                    "This usually indicates a network connectivity issue or server problem."
+                )
+                logger.error("Troubleshooting steps:")
+                logger.error("1. Check your internet connection")
+                logger.error(
+                    f"2. Verify the homeserver URL is correct: {matrix_homeserver}"
+                )
+                logger.error("3. Ensure the Matrix server is online and accessible")
+                logger.error("4. Check if your credentials are still valid")
+
+            raise ConnectionError(f"Matrix sync failed: {error_type} - {error_details}")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)